config/v1/types.go

```diff
@@ -80,6 +83,96 @@ type ConfigMapReference struct {
 	Key string `json:"filename,omitempty"`
 }
 
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Build holds cluster-wide information on how to handle builds. The canonical name is `cluster`
+type Build struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec holds user-settable values for the build controller configuration
+	Spec BuildSpec `json:"spec,omitempty"`
+}
+
+type BuildSpec struct {
+	// AdditionalTrustedCA is a reference to a ConfigMap containing additional CAs that
+	// should be trusted for image pushes and pulls during builds.
+	AdditionalTrustedCA ConfigMapReference `json:"additionalTrustedCA,omitempty"`
```

Comment:
`	// +optional`
so that openapi is generated correctly.  do that throughout on optional fields

Reply:
`Build.Spec` should not be optional then, correct?

Reply:
it's optional, technically you don't have to provide any config.